File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.c
int max(int a, int b) {
    return a > b ? a : b;
}

int lengthOfLongestSubstring(char* s) {
    bool map[128];
    int l = 0, maxL = 0;

    for (int r = 0; r < strlen(s); r++) {
        while (map[s[r]]) {
            map[s[l]] = false;
            l++;
        }
        map[s[r]] = true;;
        maxL = max(maxL, r -l + 1);
       }
       return maxL;
}

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
       boolean[] map = new boolean[128];
       int l = 0, maxL = 0;

       for (int r = 0; r < s.length(); r++) {
        while (map[s.charAt(r)]) {
            map[s.charAt(l)] = false;
            l++;
        }
        map[s.charAt(r)] = true;;
        maxL = Math.max(maxL, r -l + 1);
       }
       return maxL;
    }
}